BUILD

```diff
@@ -0,0 +1,39 @@
+# Description:
+# Tools for building a Catalog by Convention.
+
+licenses(["notice"])  # Apache 2.0
+
+exports_files(["LICENSE"])
+
+objc_library(
+    name = "CatalogByConvention",
+    srcs = glob([
+        "src/*.m",
+        "src/private/*.m",
+    ]),
+    hdrs = glob([
+        "src/*.h",
+        "src/private/*.h",
+    ]),
+    includes = ["src"],
+    visibility = ["//visibility:public"],
+    copts = [
+        "-Wall",  # Standard known-to-be-bugs warnings.
+        "-Wcast-align",  # Casting a pointer such that alignment is broken.
+        "-Wconversion",  # Numeric conversion warnings.
+        "-Wdocumentation",  # Documentation checks.
+        "-Werror",  # All warnings as errors.
+        "-Wextra",  # Many useful extra warnings.
+        "-Wimplicit-atomic-properties",  # Dynamic properties should be non-atomic.
+        "-Wmissing-prototypes",  # Global function is defined without a previous prototype.
+        "-Wno-error=deprecated",  # Deprecation warnings are never errors.
+        "-Wno-error=deprecated-implementations",  # Deprecation warnings are never errors.
+        "-Wno-sign-conversion",  # Do not warn on sign conversions.
+        "-Wno-unused-parameter",  # Do not warn on unused parameters.
+        "-Woverlength-strings",  # Strings longer than the C maximum.
+        "-Wshadow",  # Local variable shadows another variable, parameter, etc.
+        "-Wstrict-selector-match",  # Compiler can't figure out the right selector.
+        "-Wundeclared-selector",  # Compiler doesn't see a selector.
+        "-Wunreachable-code",  # Code will never be reached.
```

Comment:
Am wondering if we can share these warnings somewhere? Perhaps as a separate bazel config file in foundation?

@ajsecord @randallli  thoughts?

Reply:
I'll probably explore pulling this out in a follow-up change.

Reply:
sgtm